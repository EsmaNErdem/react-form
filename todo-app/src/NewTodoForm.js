import React, { useState } from "react";

const NewTodoForm = ({addTask}) => {
    const INITIAL_STATE = {task: ""}
    const [formData, setFormData] = useState(INITIAL_STATE)

    const handleSubmit = e => {
        e.preventDefault()
        addTask(formData)
        setFormData(INITIAL_STATE)
    }

    const handleChange = e => {
        const { name, value } = e.target
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="task">New Todo Task:</label>
            <input
                id="task"
                name="task"
                type="text"
                value={formData.task}
                onChange={handleChange}
            />
            <button>Add task!</button>
        </form>
    )
}

export default NewTodoForm;