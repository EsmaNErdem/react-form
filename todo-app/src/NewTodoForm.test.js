import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

test("renders without crashing", () => {
    render(<NewTodoForm />);
});

test("it renders and matches with snaphot", () => {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});