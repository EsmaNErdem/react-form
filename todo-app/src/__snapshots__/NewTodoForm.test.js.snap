// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches with snaphot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      New Todo Task:
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button>
      Add task!
    </button>
  </form>
</DocumentFragment>
`;
